adoc/chapters/programming_interface.adoc

```diff
@@ -2906,9 +2906,12 @@ function.
 
 All member functions of the [code]#queue# class are synchronous and errors
 are handled by throwing synchronous SYCL exceptions. The [code]#submit#
-member function schedules <<command-group,command groups>> asynchronously, so any errors
-in the submission of a <<command-group>> are handled by throwing synchronous
-SYCL exceptions. Any exceptions from the <<command-group>> after it has
+member function synchronously invokes the provided
+<<command-group-function-object>> (as described in
+<<sec:command-groups-exec-order>>) in the calling thread, thereby scheduling a
+<<command-group>> for asynchronous execution. Any error in the submission of a
+<<command-group>> is handled by throwing a synchronous SYCL exceptions.
+Any exceptions from the <<command-group>> after it has
```

Comment:
```suggestion
Any errors from the <<command-group>> after it has
```

I think we mean errors more generally here, right?  For example, an error from the underlying backend could cause the asynchronous error handler to be called.  An exception from the application's host task could also cause the asynchronous error handler to be called, but we say that in the definition of host tasks.

Notably, exceptions in other commands are **not** guaranteed to call the asynchronous error handler (because exception are not allowed in device code).

Reply:
Right. Not really related to this PR but I guess we can do a drive-by clarification :)